### PR TITLE
Add mission / quest NATION.MISSION / AREA.QUEST table references

### PR DESCRIPTION
**Pull request by ibm2431**
`I was about to muse on the PR that since lua tables are by reference, if we wanted NATION.MISSION to also be a valid option, could assign dsp.mission.id.NATION = dsp.mission.id.nation in the global without losing any real performance (since we wouldn't be creating a second copy in memory)`

### CODE REVIEW

**Comment by zynjec** [comment]
That's duplicating some pretty big tables for not much reason.

**Comment by ibm2431** [comment]
https://www.lua.org/pil/2.5.html

**Comment by zynjec** [comment]
Huh, guess I'm just retarded.

**Comment by ibm2431** [comment]
Nah, Lua is pretty weird.

**Comment by Cryptexx** [comment]
He's right in that it's referenced. 

I often use local t = some.stupidly.deep.hierarchy.of.a.table and it doesn't create a deep copy of the data.  Just stores the reference to that table in local t. As an interesting note too, I found that localizing tables and functions actually speed things up when it's referenced multiple times in the same function.

Kinda like using local ran = math.random is better than calling math.random multiple times in sequence.

In this case though, no performance gain or loss, just naming conventions/preferences.

**Comment by ibm2431** [comment]
To be fair, I can't state _how_ useful it would be.

Mainly just a, "hey, we could do this if we're concerned about case sensitivity". And so here we are.

My best articulation would be:
"Since we use UPPERCASE for the final suffix for nations and mission/quest log areas (`dsp.nation.BASTOK`, `getQuestStatus(JEUNO,`) , it's easy to get into a habit of capitalizing those as well. This way there wouldn't be any issues if that happens either through scripter habit or preference."

Of course, on the _other_ hand:
"We should stick to the same convention for all tabled enums so there isn't confusion for people using _other_ tables."

This was a 5-minute commit; I wouldn't be offended if people decide one way or the other...!

**Comment by nfaletra** [comment]
Ha. Reading is hard. I lean towards consistency over supporting multiple formats. I haven’t looked at these lua files in awhile, is there a reason it’s defined in a lower case manner to begin with?

**Comment by ibm2431** [comment]
Just because the lower.lower.FINAL is the style elsewhere (Nations, Key Items, Magic Elements, etc).

For mission IDs, areas being lower does make the FINAL_REFERENCE visually prominent though. It might just be a "where do we want to apply visual weight" thing:

"If someone is looking through a script, do we want to draw their attention to `mission.WINDURST.THE_HEART_OF_THE_MATTER` or `mission.windurst.THE_HEART_OF_THE_MATTER`?"

**Comment by TeoTwawki** [comment]
we had a convention collision during some major renovations. Prior to the mass tabling of all the things the global variables were all in cap. all the tables were done in lowercase and new non tabled variables are being done lowercase... so new things in general started being done full lowercase and mingling has begun. We didn't set a prominent guideline on how to do this stuff soon enough and its catching up to us now.

unrelated: also stuff became very wide thanks to nested tables, [which has lead to some ultra short variables to compensate](https://github.com/DarkstarProject/darkstar/blob/817225f1eda7dc68ada684090183a4ba96dafb04/scripts/globals/bcnm.lua#L443) (2 letter variables bother me, because I expect them to accidentally get used twice at some point)

**Comment by takhlaq** [comment]
SNAKE_CASE_CAPS for enums since we have them all that way in cpp already

**Comment by TeoTwawki** [comment]
> SNAKE_CASE_CAPS for enums since we have them all that way in cpp already

but now we have ENUMs in.a.table.

**Comment by TeoTwawki** [comment]
tl;dr snake has a fat ass
